qutebrowser/completion/models/miscmodels.py

```diff
@@ -257,3 +239,47 @@ def delete_cur_item(self, completion):
         tabbed_browser = objreg.get('tabbed-browser', scope='window',
                                     window=int(win_id))
         tabbed_browser.on_tab_close_requested(int(tab_index) - 1)
+
+
+class BindCompletionModel(base.BaseCompletionModel):
+
+    """A CompletionModel filled with all bindable commands and descriptions."""
+
+    # https://github.com/The-Compiler/qutebrowser/issues/545
+    # pylint: disable=abstract-method
+
+    COLUMN_WIDTHS = (20, 60, 20)
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        cmdlist = _get_cmd_completions(include_hidden=True,
+                                       include_aliases=True)
+        cat = self.new_category("Commands")
+        for (name, desc, misc) in cmdlist:
+            self.new_item(cat, name, desc, misc)
+
+def _get_cmd_completions(include_hidden, include_aliases, prefix=''):
+    """Get a list of completions info for commands, sorted by name.
+
+    Args:
+        include_hidden: True to include commands annotated with hide=True.
+        include_aliases: True to include command aliases.
+        prefix: String to append to the command name.
+
+    Return: A list of tuples of form (name, description, bindings).
+    """
+    assert cmdutils.cmd_dict
+    cmdlist = []
+    cmd_to_keys = objreg.get('key-config').get_reverse_bindings_for('normal')
+    for obj in set(cmdutils.cmd_dict.values()):
+        if ((not obj.debug or objreg.get('args').debug) and
+             not obj.deprecated and
+             (include_hidden or not obj.hide)):
+            bindings = ', '.join(cmd_to_keys[obj.name])
```

Comment:
I was really confused why this worked and didn't raise `KeyError` for unbound commands - I think it does before `get_reverse_bindings_for` uses a `collections.defaultdict(list)` to build up things and then returns that.

I think it would be cleaner for `get_reverse_bindings_for` to return a normal `dict`, and then we'd have explicitly do `cmd_to_keys.get(obj.name, [])` here. What do you think?